frontend/src/pages/DegreeWizard/steps/SpecialisationStep.jsx

```diff
@@ -3,51 +3,49 @@ import { useDispatch, useSelector } from "react-redux";
 import { animated, useSpring } from "@react-spring/web";
 import { Button, Menu, Typography } from "antd";
 import axios from "axios";
-import { addMajor, removeMajor } from "reducers/degreeSlice";
+import { addSpecialisation, removeSpecialisation } from "reducers/degreeSlice";
 import springProps from "./spring";
 import "./steps.less";
 
 const { Title } = Typography;
-const SpecialisationStep = ({ incrementStep, currStep }) => {
+const SpecialisationStep = ({ incrementStep, type }) => {
   const dispatch = useDispatch();
-  const { programCode, majors } = useSelector((store) => store.degree);
+  const { programCode, specs } = useSelector((store) => store.degree);
   const [options, setOptions] = useState({ someProgramName: { specs: { major: "major data" } } });
 
   const fetchAllSpecialisations = useCallback(async () => {
-    const res = await axios.get(`/programs/getMajors/${programCode}`);
-    setOptions(res.data.majors);
-  }, [programCode]);
+    const res = await axios.get(`/specialisations/getSpecialisations/${programCode}/${type}`);
+    setOptions(res.data.spec);
+  }, [programCode, type]);
 
   useEffect(() => {
     if (programCode !== "") fetchAllSpecialisations();
-  }, [fetchAllSpecialisations, programCode]);
+  }, [fetchAllSpecialisations, programCode, type]);
 
   const props = useSpring(springProps);
   return (
     <animated.div style={props} className="steps-root-container">
       <div className="steps-heading-container">
         <Title level={4} className="text">
-          What do you major in?
+          What are your {type}?
         </Title>
-        {majors.length > 0 && currStep === 3 && (
-          <Button type="primary" onClick={incrementStep}>
-            Next
-          </Button>
-        )}
+        <Button type="primary" onClick={incrementStep}>
+          Next
+        </Button>
       </div>
 
       <Menu
         className="degree-specialisations"
-        onSelect={(e) => dispatch(addMajor(e.key))}
-        onDeselect={(e) => dispatch(removeMajor(e.key))}
-        selectedKeys={majors}
+        onSelect={(e) => dispatch(addSpecialisation(e.key))}
+        onDeselect={(e) => dispatch(removeSpecialisation(e.key))}
+        selectedKeys={specs}
         defaultOpenKeys={["0"]}
         mode="inline"
       >
         {Object.keys(options).map((sub, index) => (
           <Menu.SubMenu
             key={index}
-            title={`Majors for ${sub}`}
+            title={`${type} for ${sub}`}
```

Comment:
Ooof anyway to capitalise the first letter? i.e. Instead of `majors for Computer Science`, it could be `Majors for Computer Science`?